src/shape/shape.rs

```diff
@@ -282,6 +282,9 @@ pub trait Shape: RayCast + PointQuery + DowncastSync {
     #[cfg(feature = "std")]
     fn clone_box(&self) -> Box<dyn Shape>;
 
+    #[cfg(feature = "std")]
+    fn clone_scaled(&self, scale: &Vector<Real>, num_subdivisions: u32) -> Option<Box<dyn Shape>>;
```

Comment:
This is lacking some documentation. The documentation should mention the fact that, in some cases, the resulting shape doesn’t have the same type as `Self`. For example, if a non-uniform scale is provided and `Self` as a `Ball`, then the result will be discretized (based on the `num_subdivisions` parameter) as a `ConvexPolyhedron` (in 3D) or `ConvexPolygon` (in 2D).

Because of that potential type change, I’m not sure that `clone_scaled` is the best name here (since `Clone` typically return the same type). Does calling it just `scaled` clash with the inherent functions `scaled` in any way when the users calls it?

Reply:
I've updated the docs.

I really don't want to create a naming ambiguity like that. Let's name it `dyn_scaled` instead then.

Reply:
Thank you. I did one last bit of renaming: `clone_box` is now `clone_dyn` and `dyn_scaled` is `scale_dyn`. So that the prefix/suffix are consistent.